File: Lista IV (JAVA) - Aula 10/J2_04/Produto.java
public abstract class Produto {
    private String nome;
    private double preco;

    Produto(){}

    Produto(String nome, double preco){
        this.setNome(nome);
        this.setPreco(preco);
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNome() {
        return nome;
    }

    public void setPreco(double preco) {
        this.preco = preco;
    }
    
    public double getPreco() {
        return preco;
    }

    public String toString(){
        String s = this.nome + String.format(" (R$%.2f) - ", this.preco);
        
        return s;
    }
}

File: Lista V (C++) - Aula 22/C1_07/Produto.cpp
#include <iostream>
#include "Produto.h"
using namespace std;

//Construtor
Produto::Produto(int cod, const string& nome, double preco){
    this->cod = cod;
    this->nome = nome;
    this->preco = preco;
}

//GETs 
const string& Produto::getNome() const{
    return this->nome;
}

double Produto::getPreco() const{
    return this->preco;
}

//Métodos
void Produto::print(ostream &out) const{ 
    out << nome << " (R$ " << preco << ")"; //coloquei em "protected" pois faz esses acessos diretos aos atributos
}

ostream& operator<<(ostream& out, const Produto& p){
    p.print(out);
    return out;
}

bool operator==(const Produto& a, const Produto& b) {
	return a.cod == b.cod;
}